src/backends/github/API.js

```diff
@@ -78,7 +78,11 @@ export default class API {
       if (contentType && contentType.match(/json/)) {
         return this.parseJsonResponse(response);
       }
-      return response.text();
+      const text = response.text();
+      if (!response.ok) {
```

Comment:
@Benaiah Is there any problem internally if we reject on not-OK responses for all content-types? We do on JSON responses already (see `parseJsonResponse`).

Reply:
@tech4him1 yeah - I actually thought it was supposed to be doing so already. If the caller is expecting a non-ok response it can catch the `APIError`.